src/Symfony/Component/HttpKernel/Attribute/MapSessionParameter.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Attribute;
+
+use Symfony\Component\HttpKernel\Controller\ArgumentResolver\SessionParameterValueResolver;
+
+/**
+ * Can be used to pass a session parameter to a controller argument.
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class MapSessionParameter extends ValueResolver
+{
+    public function __construct(
+        string $resolver = SessionParameterValueResolver::class,
```

Comment:
ah, one thing: let's add a `name` parameter to allow using a different name than the controller's parameter:
```
@param string|null $name The name of the session parameter; if null, the name of the argument in the controller will be used
```